src/CaptureFile/CaptureFileOutputStreamTest.cpp

```diff
@@ -0,0 +1,83 @@
+
+namespace orbit_capture_file {
+
+TEST(CaptureFileOutputStream, Smoke) {
+  // TODO(http://b/180574275): Replace this with temporary_file once it becomes available
+  const char* temp_file_name = std::tmpnam(nullptr);
+
+  auto output_stream_or_error = CaptureFileOutputStream::Create(temp_file_name);
+  ASSERT_TRUE(output_stream_or_error.has_value()) << output_stream_or_error.error().message();
+
+  std::unique_ptr<CaptureFileOutputStream> output_stream =
+      std::move(output_stream_or_error.value());
+
+  constexpr const char* kAnswerDescription =
+      "Answer to the Ultimate Question of Life, The Universe, and Everything";
+
+  orbit_grpc_protos::ClientCaptureEvent event;
+  orbit_grpc_protos::InternedString* interned_string = event.mutable_interned_string();
+  interned_string->set_key(42);
+  interned_string->set_intern(kAnswerDescription);
+
+  output_stream->WriteCaptureEvent(event);
+  output_stream->Close();
+
+  ErrorMessageOr<std::string> file_content_or_error = orbit_base::ReadFileToString(temp_file_name);
+  ASSERT_TRUE(file_content_or_error.has_value()) << file_content_or_error.error().message();
+
+  const std::string& file_content = file_content_or_error.value();
+
+  ASSERT_GT(file_content.size(), 24);
+  ASSERT_EQ(file_content.substr(0, 4), kFileSignature);
+  uint64_t capture_section_offset = 0;
+  memcpy(&capture_section_offset, file_content.data() + 8, sizeof(capture_section_offset));
+  ASSERT_EQ(capture_section_offset, 24);
+
+  google::protobuf::io::ArrayInputStream input_stream(
+      file_content.data() + capture_section_offset,
+      static_cast<int>(file_content.size() - capture_section_offset));
+  google::protobuf::io::CodedInputStream coded_input_stream(&input_stream);
+
+  orbit_grpc_protos::ClientCaptureEvent event_from_file;
+  event_from_file.ParseFromCodedStream(&coded_input_stream);
+
+  ASSERT_EQ(event_from_file.event_case(), orbit_grpc_protos::ClientCaptureEvent::kInternedString);
+  EXPECT_EQ(event_from_file.interned_string().key(), 42);
+  EXPECT_EQ(event_from_file.interned_string().intern(), kAnswerDescription);
+}
+
+TEST(CaptureFileOutputStream, WriteAfterClose) {
+  // TODO(http://b/180574275): Replace this with temporary_file once it becomes available
+  const char* temp_file_name = std::tmpnam(nullptr);
+
+  auto output_stream_or_error = CaptureFileOutputStream::Create(temp_file_name);
+  ASSERT_TRUE(output_stream_or_error.has_value()) << output_stream_or_error.error().message();
+
+  std::unique_ptr<CaptureFileOutputStream> output_stream =
+      std::move(output_stream_or_error.value());
+
+  output_stream->Close();
+
+  constexpr const char* kAnswerDescription =
+      "Answer to the Ultimate Question of Life, The Universe, and Everything";
+
+  orbit_grpc_protos::ClientCaptureEvent event;
+  orbit_grpc_protos::InternedString* interned_string = event.mutable_interned_string();
+  interned_string->set_key(42);
+  interned_string->set_intern(kAnswerDescription);
```

Comment:
Maybe you can extract this into a function as you use the same in the other test.

Reply:
done.